packages/xml-views-completion/src/api.ts

```diff
@@ -36,7 +37,36 @@ export function getXMLViewCompletions(
       includes(allowedVisibility, _.ui5Node.visibility)
   );
 
-  return publicAndProtectedSuggestions;
+  const filteredSuggestions = filterBySettings(
```

Comment:
`filterBySettings` is implemented using `remove` which **mutates** its argument.
This means that after this statement `filteredSuggestions === publicAndProtectedSuggestions`

So if we add breakpoints during debug it would not be trivial be follow the changes/transformations.

Could we use `filter` or `reject` which create a new array?

Reply:
Changed to `reject`